lambdas/report-user-identity/src/main/java/uk/gov/di/ipv/core/reportuseridentity/ReportUserIdentityHandler.java

```diff
@@ -0,0 +1,328 @@
+        } catch (StopBeforeLambdaTimeoutException e) {
+            LOGGER.error(LogHelper.buildErrorMessage("Stopping as lambda about to timeout.", e));
+        } finally {
+            LOGGER.info(LogHelper.buildLogMessage("Writing output with result summary."));
+            objectMapper.enable(SerializationFeature.INDENT_OUTPUT);
+            objectMapper.writeValue(outputStream, reportProcessingResult.build());
+        }
+    }
+
+    private void scanToExtractUniqueUserIdFromTacticalStore(
+            ReportProcessingResult.ReportProcessingResultBuilder reportProcessingResult,
+            ReportProcessingRequest reportProcessingRequest,
+            Context context)
+            throws BatchDeleteException, StopBeforeLambdaTimeoutException {
+        if (reportProcessingRequest.continueUniqueUserScan()) {
+            LOGGER.info(
+                    LogHelper.buildLogMessage("Retrieving userIds from tactical store db table."));
+
+            for (var page :
+                    vcStoreItemScanDynamoDataStore.getItems(
+                            reportProcessingRequest.tacticalStoreLastEvaluatedKey(),
+                            ATTR_NAME_USER_ID)) {
+
+                var reportUserIdentities =
+                        page.items().stream()
+                                .map(VcStoreItem::getUserId)
+                                .distinct()
+                                .map(
+                                        usrId ->
+                                                new ReportUserIdentityItem(
+                                                        usrId, null, null, null, null))
+                                .toList();
+                reportUserIdentityScanDynamoDataStore.createOrUpdate(reportUserIdentities);
+
+                reportProcessingResult.tacticalStoreLastEvaluatedKey(page.lastEvaluatedKey());
+
+                checkLambdaRemainingExecutionTime(context);
+            }
+            LOGGER.info(
+                    LogHelper.buildLogMessage(
+                            "Tactical storage scan completed to get unique users."));
+            reportProcessingResult.tacticalStoreLastEvaluatedKey(null);
+        }
+    }
+
+    private void processUsersToFindLOCAndUpdateDb(
+            ReportProcessingRequest reportProcessingRequest,
+            ReportProcessingResult.ReportProcessingResultBuilder reportProcessingResult,
+            Context context)
+            throws CredentialParseException, HttpResponseExceptionWithErrorBody, ParseException,
+                    BatchDeleteException, StopBeforeLambdaTimeoutException {
+        if (reportProcessingRequest.continueUserIdentityScan()) {
+            LOGGER.info(
+                    LogHelper.buildLogMessage(
+                            "Processing unique users to update their aggregate identities."));
+
+            List<ReportUserIdentityItem> totalReportUserIdentityItems = new ArrayList<>();
```

Comment:
This is still trying to collect a list of all user identities in-memory - can we remove this.